src/Shared/PackageManagers/GitHubProjectManager.cs

```diff
@@ -140,7 +140,7 @@ public override async Task<IEnumerable<string>> DownloadVersion(PackageURL purl,
                     });
                 }
                 PackageURL purlNoVersion = new(purl!.Type, purl.Namespace, purl.Name, null, purl.Qualifiers, purl.Subpath);
-                foreach (string v in EnumerateVersions(purlNoVersion).Result)
+                foreach (string v in EnumerateVersionsAsync(purlNoVersion).Result)
```

Comment:
```suggestion
                await foreach (string v in EnumerateVersionsAsync(purlNoVersion))
```

Reply:
Same situation as https://github.com/microsoft/OSSGadget/pull/327#discussion_r849927142

Reply:
I can do `foreach (string v in await EnumerateVersionsAsync(purlNoVersion))` instead